share/functions/__fish_complete_mysql.fish

```diff
@@ -0,0 +1,24 @@
+function __fish_complete_mysql
+    set -l command $argv[1]
+    function __fish_get_mysql_cmd
+        argparse -i 'u/user=' 'P/port=' 'h/host=' 'p/password=?' 'S/socket=' -- (commandline -po)
+        set -l mysql_cmd mysql
+        for flag in u P h S
+            if set -q _flag_$flag
+                set -l flagvar _flag_$flag
+                set -a mysql_cmd -$flag $$flagvar
+            end
+        end
+        if [ -n "$_flag_p" ]
+            set -a mysql_cmd -p$_flag_p
+        end
+        echo $mysql_cmd
+    end
+
+    function __fish_complete_mysql_databases
+        echo 'show databases' | eval (__fish_get_mysql_cmd) 2>/dev/null
```

Comment:
Why not just... run the command?

```fish
__fish_mysql_cmd 'show databases'
```

and then make `__fish_mysql_cmd` just run the mysql command instead of `echo`ing it?

Reply:
Or better yet, just do the `argparse` directly in this function and run it directly, no need for the indirection.

Reply:
Giving the query to `__fish_mysql_cmd` is a good idea. 

Running argparse inside the function is not a good idea in my opinion. Since in the future we possibly might add table completion or something, which would need exactly the same argument parsing. So I have split that functionality in advance to accommodate for that.